scipy/special/_basic.py

```diff
@@ -2791,6 +2791,12 @@ def perm(N, k, exact=False):
 
     """
     if exact:
+        if not (isscalar(N) and isscalar(k)):
+            raise ValueError("`N` and `k` must scalar integers be with `exact=True`.")
+        if not (floor(N) == N and floor(k) == k):
+            msg = ("Non-integer `N` and `k` with `exact=True` is deprecated and will "
+                   "raise an error in SciPy 1.17.0.")
```

Comment:
Seems like there's still time to get this into 1.14, so this could become an error in 1.16?